parsl/dataflow/task_status_poller.py

```diff
@@ -0,0 +1,67 @@
+import logging
+import parsl  # noqa F401 (used in string type annotation)
+import time
+from typing import Dict, Any, Sequence
+from typing import List  # noqa F401 (used in type annotation)
+
+from parsl.dataflow.executor_status import ExecutorStatus
+from parsl.dataflow.job_error_handler import JobErrorHandler
+from parsl.dataflow.strategy import Strategy
+from parsl.executors.base import ParslExecutor
+from parsl.providers.provider_base import JobStatus
+
+
+logger = logging.getLogger(__name__)
+
+
+class PollItem(ExecutorStatus):
+    def __init__(self, executor: ParslExecutor):
+        self._executor = executor
+        self._interval = executor.status_polling_interval
+        self._last_poll_time = 0.0  # type: float
+        self._status = {}  # type: Dict[Any, JobStatus]
+
+    def _should_poll(self, now: float):
+        return now >= self._last_poll_time + self._interval
+
+    def poll(self, now: float):
+        if self._should_poll(now):
+            logging.debug("Polling {}".format(self._executor))
```

Comment:
s/logging/logger/

Reply:
There's inline editing. Might be quicker than comment, wait for me to see, me edit, etc.